lib/winston/transports/file.js

```diff
@@ -88,6 +89,7 @@ module.exports = class File extends TransportStream {
     this._opening = false;
     this._ending = false;
 
+    if (this.dirname) this._createLogDirIfNotExist(this.dirname);
```

Comment:
@adoyle-h Why do we need `if (this.dirname)`?

Reply:
For robustness. `_createLogDirIfNotExist` must be passed with a string parameter. Maybe `assert` is better, but `assert` will break the execution down.
So if `this.dirname` is undefined, just not create directory.